File: Data Structures/Stack/stack.py
#stack implementation using list
stack = []

# append() to push to stack
stack.append('a')
stack.append('b')
stack.append('c')


print('Initial stack')
print(stack)

# pop() to pop element in LIFO order
print('\nElements popped form stack:')
print(stack.pop())
print(stack.pop())

print('\nStack after popping')
print(stack)


#===============================================================

from collections import deque

stack = deque()

# push()
stack.append('a')

# pop()
print(stack.pop())

#===============================================================

from queue import LifoQueue

stack = LifoQueue(maxsize=3)

# push()
stack.put('a')

# pop()
stack.get()

# size()
stack.qsize()

# isFull()
stack.full()

# isEmpty()
stack.empty()

File: Data Structures/Stack/stack.js
const stack = []

// pus
stack.push('mazda');

console.log(stack);

stack.push( 'fiat', 'volvo', 'volkswagen' ) 

console.log(stack)

// pop() - last item - in LIFO order
stack.pop()
console.log(stack)

stack.pop('volvo')
console.log(stack)

stack.pop(1)
console.log(stack)
